src/Illuminate/Validation/ValidationRuleParser.php

```diff
@@ -96,17 +95,11 @@ protected function explodeExplicitRule($rule, $attribute)
             return Arr::wrap($this->prepareRule($rule, $attribute));
         }
 
-        $rules = [];
-
-        foreach ($rule as $value) {
-            if ($value instanceof Date) {
-                $rules = array_merge($rules, explode('|', (string) $value));
-            } else {
-                $rules[] = $this->prepareRule($value, $attribute);
-            }
-        }
-
-        return $rules;
+        return array_map(
+            [$this, 'prepareRule'],
+            $rule,
+            array_fill((int) array_key_first($rule), count($rule), $attribute)
```

Comment:
The `(int)` cast doesn't make sense. You probably expect the key to be of type int anyway, but if it for some reason is not, casting a potentially non-empty string to int won't be very helpful, would it?